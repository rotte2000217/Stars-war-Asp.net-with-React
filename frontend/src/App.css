.App {
  text-align: center;
  font-family: 'Roboto', sans-serif;
}

.Login {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #000000;
  min-height: 80vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.App-Theme {
  background-color: #000000;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
h3
{
  color: #FFD700;
}
.hidden { display:none; }
