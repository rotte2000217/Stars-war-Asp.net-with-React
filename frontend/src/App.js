import React, {Component} from 'react';
import logo from './Star_Wars_Logo.svg.png';
import Simple_Question from './simplequestion'
import List_Question_3 from './question3'
import List_Question_4 from './question4'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      Q1_Question: "Which of all Star Wars movies has the longest opening crawl ?",
      Q1_Ans: "",
      Q2_Question: "What character (person) appeared in most of the Star Wars films?Â ",
      Q2_Ans: ""
    };
  }
  render()
  {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to the prototype interactive test</h1>
        </header>
        <div className="App-Theme">
        <h2>{this.state.Q1_Question}</h2>
        <h3>{this.state.Q1_Ans}</h3>
        <h2>{this.state.Q2_Question}</h2>
        <h3>{this.state.Q2_Ans}</h3>
        <List_Question_3 />
        <List_Question_4 />
        </div>
      </div>
    );
  };
  componentWillMount() {
    document.title = 'Dashboard';
    fetch("/api/OpeningCrawl", {
      method: 'GET'
  })
  .then(res => res.json())
  .then((dat) => {
  var data = JSON.parse(dat);
  if(data != null && data.length > 0 )
  {
    this.setState({Q1_Ans: data[0].title});
  }
 })

 fetch("/api/CharacterAppearance", {
  method: 'GET'
  })
  .then(res => res.json())
  .then((dat) => {
  var data = JSON.parse(dat);
  if(data != null && data.length > 0 )
  {
    this.setState({Q2_Ans: data[0].name});
  }
 })
}
}
export default App;