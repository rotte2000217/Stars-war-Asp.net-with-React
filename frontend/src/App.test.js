import React from "react";
import renderer from "react-test-renderer";
import App from "./App";
import Question3 from "./question3";
import Question4 from "./question4";

import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

describe("App", () => {
  test("snapshot renders App", () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Question3", () => {
  test("snapshot renders Question3", () => {
    const component = renderer.create(<Question3 />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Question4", () => {
  test("snapshot renders Question4", () => {
    const component = renderer.create(<Question4 />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test('Q3', () => {
  const wrapper = mount(<Question3 />)
  expect(wrapper.instance().state.Question).toBe('What species appeared in the most  number of Star Wars films?');
});

test('Q4', () => {
  const wrapper = mount(<Question4 />)
  expect(wrapper.instance().state.Question).toBe('What planet in Star Wars universe provided largest number of vehicle pilots?');
});

describe('<h2 />', () => {
  it('renders App h2', () => {
    const wrapper = shallow(<App />);
    const elem = wrapper.find('h2');
    expect(elem.length).toBe(2);
  });
});

describe('<div />', () => {
  it('renders App div', () => {
    const wrapper = shallow(<App />);
    const elem = wrapper.find('div');
    expect(elem.length).toBe(2);
  });
});

describe('<h3 />', () => {
  it('renders h3 count', () => {
    const wrapper = shallow(<App />);
    const elem = wrapper.find('h3');
    expect(elem.length).toBe(2);
  });
});

 it('able to find an html element Q3', () => {
  const wrapper = shallow(<Question3 />);
  expect(wrapper.find('h2').html()).toContain('What species appeared');
 });

 it('able to find an html element Q4', () => {
  const wrapper = shallow(<Question4 />);
  expect(wrapper.find('h2').html()).toContain('What planet in Star');
 });

 it('able to find an html element by class App', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find('.App-logo').html())
   .toContain('Star_Wars_Logo');
 });

 it('able to find an html element by class Q3', () => {
  const wrapper = shallow(<Question3 />);
  expect(wrapper.find('.App-Theme').html())
   .toContain('What species appeared');
 });

 it('able to find an html element by class Q4', () => {
  const wrapper = shallow(<Question4 />);
  expect(wrapper.find('.App-Theme').html())
   .toContain('What planet in Star');
 });

 it('able to find an html element by id Q4', () => {
  const wrapper = shallow(<Question4 />);
  expect(wrapper.find('#QuestionText').html())
   .toContain('What planet in Star');
 });

 it('able to find a React Component Q3', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Question3).length).toBe(1);
 });

 it('able to find a React Component Q4', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Question4).length).toBe(1);
 });
