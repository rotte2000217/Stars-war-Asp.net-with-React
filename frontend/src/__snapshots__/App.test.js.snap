// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders App 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="Star_Wars_Logo.svg.png"
    />
    <h1
      className="App-title"
    >
      Welcome to the prototype interactive Test
    </h1>
  </header>
  <div
    className="App-Theme"
  >
    <h2>
      Which of all Star Wars movies has the longest opening crawl?
    </h2>
    <h3>
      
    </h3>
    <h2>
      What character (person) appeared in most of the Star Wars films?
    </h2>
    <h3>
      
    </h3>
    <div
      className="App-Theme"
    >
      <h2
        id="QuestionText"
      >
        What species appeared in the most  number of Star Wars films?
      </h2>
    </div>
    <div
      className="App-Theme"
    >
      <h2
        id="QuestionText"
      >
        What planet in Star Wars universe provided largest number of vehicle pilots?
      </h2>
    </div>
  </div>
</div>
`;

exports[`Question3 snapshot renders Question3 1`] = `
<div
  className="App-Theme"
>
  <h2
    id="QuestionText"
  >
    What species appeared in the most  number of Star Wars films?
  </h2>
</div>
`;

exports[`Question4 snapshot renders Question4 1`] = `
<div
  className="App-Theme"
>
  <h2
    id="QuestionText"
  >
    What planet in Star Wars universe provided largest number of vehicle pilots?
  </h2>
</div>
`;
